### ISSUE TITLE
About finetune.py

### ISSUE THREAD

**Issue by heguohao0728** [JiaRenChang/RealtimeStereo#6]
Hello, Great work!
I want to train my own module but I met some problems. Could you please give me some advice. Really appreciated.
In RTS, the **finetune.py** is same as PSM-Net, and it will encounter so problem.
Like:

`if args.model == 'stackhourglass':`
    `model = stackhourglass(args.maxdisp)`
`elif args.model == 'basic':`
`    model = basic(args.maxdisp)`
`else:`
`    print('no model')`

if I use arg.model = RTS..or something, it will report a bug.
and in submodule.py:

`class disparityregression(nn.Module):`
    `def __init__(self, maxdisp):`
        `super(disparityregression, self).__init__()`
        `self.disp = torch.Tensor(np.reshape(np.array(range(`**maxdisp +1**`)), [1,` **maxdisp+1**`, 1, 1])).cuda()`

this part is different from PSM-Net, and it will enconter a size bug.
How can I do with these problems?

**Comment by heguohao0728** [comment]
What's more. 
In submodule.py, if I change the above **maxdisp + 1** to **maxdisp** and avoid the problem, it still have some issue:
`class disparityregression(nn.Module):`
    `def __init__(self, maxdisp):`
       ` super(disparityregression, self).__init__()`
       ` self.disp = torch.Tensor(np.reshape(np.array(range(maxdisp)), [1, maxdisp, 1, 1])).cuda()`

  `  def forward(self, x):`
       ` out = torch.sum(x * self.disp.data, 2, keepdim=True)`
      `  return out`
In the forward function, the "2" must be "1", or the output will have a size problem.

**Comment by heguohao0728** [comment]
When I do my change work above, the finetune.py can run correctly, but it seems the finetune.py  do not use any code in the **RT_Stereo.py**, and it might still be PSM-Net not RTS. 
Could you please give some advice about this? or specify how to train in RTS?

**Comment by Abhishekvats1997** [comment]
@heguohao0728 were you able to run inference on this one with the weights provided ?

**Comment by JiaRenChang** [comment]
@heguohao0728  @Abhishekvats1997 

The header "module" is indicated that the model is trained using nn.DataParallel().
If You do not us the nn.DataParallel(), loading the pretrained weight without this.
I will modify it as soon as possible.

**Comment by heguohao0728** [comment]
@JiaRenChang 
Thank you so much!
But I found that the weight and the sequence in pre-trained model is not the same as the model that is defined in code.
For example, the pre-trained model has **"stage4"** but the model defined in code have **"attention"** instead. 
What's the difference?

**Comment by JiaRenChang** [comment]
@heguohao0728 

I modify all of them.
You would load pretrained model without conflicts now.
Let me know if it has problems.

**Comment by heguohao0728** [comment]
@JiaRenChang 
Thank you so much! I'v tried, but my result seems not good enough:
![000027_10](https://user-images.githubusercontent.com/54935494/114492044-50ad9c00-9c4a-11eb-9e96-332fda81179d.png)
![Test_disparity](https://user-images.githubusercontent.com/54935494/114492075-5dca8b00-9c4a-11eb-9f1d-8e9bdcada648.png)
Is it right? or I am not using the pretrained model correctly?
Besides, I notice that the **attention** module in **RTS** is deleted and replaced by **stage4**. Is the stage4 means attention module?
Appriciated.

**Comment by JiaRenChang** [comment]
@heguohao0728 
Hi, I think it right. 
The proposed method has around 7% 3px error on KITTI dataset, but it runs real-time on TX2.
If you want accurate disparity map instead of inference speed, you can try PSMNet.
Yep, the stage4 means that the attention module.

**Comment by heguohao0728** [comment]
@JiaRenChang 
That help a lot! Really appreciated for your reply!

**Comment by flyxuexi** [comment]
> If you want accurate disparity map instead of inference speed, you ca

Thanks for your efforts to reproduce.when I using finetune.py,The following code seems wrong：
        if args.model == 'stackhourglass':
            output1, output2, output3 = model(imgL,imgR)
            output1 = torch.squeeze(output1,1)
            output2 = torch.squeeze(output2,1)
            output3 = torch.squeeze(output3,1)
            loss = 0.5*F.smooth_l1_loss(output1[mask], disp_true[mask], size_average=True) + 0.7*F.smooth_l1_loss(output2[mask], disp_true[mask], size_average=True) + F.smooth_l1_loss(output3[mask], disp_true[mask], size_average=True) 
        elif args.model == 'basic':
            output = model(imgL,imgR)
            output = torch.squeeze(output3,1)
            loss = F.smooth_l1_loss(output3[mask], disp_true[mask], size_average=True)

All output is about ’stackhourglass‘ and ’basic‘.Where is RTStereoNet？ 
could you please tell me how to run finetume.py correctly?
Thanks a lot

**Comment by flyxuexi** [comment]
> @JiaRenChang That help a lot! Really appreciated for your reply!

Thanks for your efforts to reproduce.when I using finetune.py,The following code seems wrong：
if args.model == 'stackhourglass':
output1, output2, output3 = model(imgL,imgR)
output1 = torch.squeeze(output1,1)
output2 = torch.squeeze(output2,1)
output3 = torch.squeeze(output3,1)
loss = 0.5F.smooth_l1_loss(output1[mask], disp_true[mask], size_average=True) + 0.7F.smooth_l1_loss(output2[mask], disp_true[mask], size_average=True) + F.smooth_l1_loss(output3[mask], disp_true[mask], size_average=True)
elif args.model == 'basic':
output = model(imgL,imgR)
output = torch.squeeze(output3,1)
loss = F.smooth_l1_loss(output3[mask], disp_true[mask], size_average=True)

All output is about ’stackhourglass‘ and ’basic‘.Where is RTStereoNet？
could you please tell me how to run finetume.py correctly?
Thanks a lo
